state/plugin.go

```diff
@@ -217,18 +237,18 @@ func getPluginBy(txn *memdb.Txn, name, svcID, routeID, consumerID string) (
 }
 
 // GetByProp returns a plugin which matches all the properties passed in
-// the arguments. If serviceID, routeID and consumerID are empty strings, then
-// a global plugin is searched.
+// the arguments. If serviceID, routeID, consumerID and consumerGroupID
+// are empty strings, then a global plugin is searched.
 // Otherwise, a plugin with name and the supplied foreign references is
 // searched.
 // name is required.
-func (k *PluginsCollection) GetByProp(name, serviceID,
-	routeID string, consumerID string,
+func (k *PluginsCollection) GetByProp(
```

Comment:
Nothing actionable for now but given that `GetByProp` has grown to have 5 params now, it became hard to understand which parameter is used for what. Especially since all of those are of type `string`.

It's also "easy to misuse": e.g. use argument in different order than expected.